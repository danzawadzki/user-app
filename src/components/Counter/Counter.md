Default counter component with all props:

```jsx
<Counter number={999} label="Label" />
```

Default counter component with out number prop (which is optional):

```jsx
<Counter label="Label" />
```
