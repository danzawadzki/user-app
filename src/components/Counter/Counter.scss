@import '../../assets/styles/colors.scss';

.Counter {
	display: flex;
	flex-direction: column;

	&__number {
		color: $brand--accent;
		font-size: 2.4rem;
		line-height: 2.9rem;
		letter-spacing: 0;
	}
	&__label {
		color: $blue--light;
		font-size: 1rem;
		line-height: 1.75rem;
		letter-spacing: 0;
	}
}
