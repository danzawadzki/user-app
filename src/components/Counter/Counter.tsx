import * as React from 'react';
import './Counter.scss';

export interface ICounter {
	/** Large counter number */
	number?: number;
	/** Small label text under the counter */
	label: string;
}

/**
 * Counter component.
 *
 * @version 1.0.0
 */
const Counter: React.StatelessComponent<ICounter> = ({ number = 0, label }) => (
	<div className="Counter">
		<span className="Counter__number">{number}</span>
		<span className="Counter__label">{label}</span>
	</div>
);

export default Counter;
