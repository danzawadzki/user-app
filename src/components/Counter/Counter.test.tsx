import React from 'react';
import { shallow } from 'enzyme';
import Counter from './Counter';

describe('Counter component: ', () => {
	// Shallow rendered component
	const num = 997;
	const label = 'Police';

	const component = shallow(<Counter number={num} label={label} />);

	it('should render match snapshot', () => {
		expect(component).toMatchSnapshot();
	});

	it('should render correctly', () => {
		expect(component.find('.Label')).toBeDefined();
	});

	it('should render with number and label', () => {
		expect(component.find('.Module__number')).not.toBe(null);
		expect(component.html()).toContain(num);
		expect(component.html()).toContain(label);
	});

	it('should render with default 0 as a counter number, if is not provided in the props', () => {
		const componentWithOutNumber = shallow(<Counter label={label} />);
		expect(componentWithOutNumber.html()).toContain(0);
		expect(componentWithOutNumber.html()).toContain(label);
	});
});
