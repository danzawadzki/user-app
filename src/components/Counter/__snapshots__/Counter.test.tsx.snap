// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component:  should render match snapshot 1`] = `
<div
  className="Counter"
>
  <span
    className="Counter__number"
  >
    997
  </span>
  <span
    className="Counter__label"
  >
    Police
  </span>
</div>
`;
